sample-3.md

```diff
@@ -0,0 +1,536 @@
+[Quick Start: Deploy "Hello World" eventing solution](todo).
+
+## Before you begin
+
+Before you begin:
+- [Install Knative Eventing](todo) and verify the health of the Knative Eventing
+  control plane.
+  
+## Description
+
+This sample is identical to [Quick Start: Deploy "Hello World" eventing solution](todo),
+but rather than using a single YAML, we will apply each piece one step at a time, 
+explaining what it does as we do so.
+
+We will create the following topology:
+
+Image from 2.
+
+- [`Pod` named `curl`](#event-producer), acting as the event producer.
+- `Broker` named `default`.
+- Two [`Trigger`s](#triggers), each with a mutually exclusivefilter pointing at distinct 
+  subscribers.
+- Two Kubernetes [`Service`s](#event-consumers), each pointing at a unique:
+    - [`Deployment`](#event-consumers), acting as the event consumer.
+
+## Helpers
+
+First we will create useful shell variables to allow us to not have to type 
+things out every time.
+
+```shell
+K_NAMESPACE=kn-eventing-step-by-step-sample
+```
+
+## Namespace
+
+Create the namespace.
+
+```shell
+kubectl create namespace $K_NAMESPACE
+```
+
+Now that the namespace has been created, we will label it to get Knative eventing
+setup. Adding this label will:
+- Create `ServiceAccount`s.
+- Create `RoleBinding`s to give permissions to those `ServiceAccount`s.
+- Create a `Broker` named `default`.
+
+```shell
+kubectl label namespace $K_NAMESPACE knative-eventing-injection=enabled
+```
+
+### Broker
+
+What is a `Broker`? It can be thought of as an event mesh. Event
+producers send events into the `Broker`. Event consumers register the kinds of events
+they are interested in with the `Broker` (via `Trigger`s, described later). The
+`Broker` ensures that every event sent to the `Broker` by event producers is sent
+to all interested event consumers.
+
+
+Let's verify that the `Broker` is in a healthy state.
+
+```shell
+kubectl -n $K_NAMESPACE get broker default
+```
+
+This should return something like:
+
+```shell
+NAME      READY   REASON   HOSTNAME                                                           AGE
+default   True             default-broker.kn-eventing-step-by-step-sample.svc.cluster.local   1m
+```
+
+We are particularly interested in the second column, `READY`. We want that value to
+be `True`. It may take a minute or two to become `True`. If it is not `True` within
+two minutes, then something has gone wrong. Hopefully the `REASON` column gives a
+useful error message. See our [Broker Debugging Guide](somewhere) to help fix the
+error.
+
+## Event Consumers
+
+The first thing we are going to do is create our event consumers. These are 
+`Deployment`s, and `Service`s that point at them. We are creating two distinct
+event consumers, so that we can show how to selectively send events to distinct
+event consumers later.
+
+Setup the '-dev' `Service` and `Deployment`.
+
+```shell
+cat << END | kubectl -n $K_NAMESPACE apply -f -
```

Comment:
I think we should explicitly call out - If using bash copy pate this. If using other shell then kubectl apply -f <link to file in this same repo>
What do you think?

Reply:
As we talked in person, I think we need to talk with the docs team on how best to do this. My personal opinion is that we should emulate Google Cloud's documentation which has tabs between different ways to do things. For example, https://cloud.google.com/compute/docs/instances/create-start-instance, not that each set of instructions has three tabs, Console, gcloud, and API. I would prefer doing something similar, where the tabs are Bash and Download YAML.

What would be _really_ cool is if we could have the markdown inject another file here, so that the download and embedded markdown are forced to stay in sync.

As for adding a call out, we should determine what the callout is, and use it consistently everywhere. For example, is it only on the first instruction per page? Or on every instruction?